server/src/modules/refresh-tokens/refresh-tokens.service.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { RefreshTokenEntity } from './refresh-token.entity';
+import { refreshTokensServiceErrorMessages } from './refresh-tokens.constants';
+import { BaseService } from 'src/common/services';
+import * as ms from 'ms';
+import { AppConfigService } from 'src/config/app-config.service';
+import { UsersService } from '../users/users.service';
+
+@Injectable()
+export class RefreshTokensService extends BaseService<RefreshTokenEntity> {
+  constructor(
+    @InjectRepository(RefreshTokenEntity)
+    private readonly refreshTokenEntityRepository: Repository<RefreshTokenEntity>,
+    private readonly appConfigService: AppConfigService,
+    private readonly usersService: UsersService,
+  ) {
+    super(refreshTokenEntityRepository, refreshTokensServiceErrorMessages);
+  }
+
+  async updateRefreshToken(userId: string, refreshToken: string) {
+    const user = await this.usersService.findOne({ id: userId });
```

Comment:
I suggest using user find condition, for being able to find not only by user id, but by other user fields (like email), bcs it may be useful in the future

const userEntity = await this.userService.findOne(user)